src/modules/workspaces/workspaces.usecase.spec.ts

```diff
@@ -311,6 +319,162 @@ describe('WorkspacesUsecases', () => {
     });
   });
 
+  describe('changeUserRole', () => {
+    it('When team does not exist, then error is throw', async () => {
+      jest.spyOn(teamRepository, 'getTeamById').mockResolvedValue(null);
+
+      await expect(
+        service.changeUserRole('workspaceId', 'teamId', {
+          userId: 'userId',
+          role: WorkspaceRole.MEMBER,
+        }),
+      ).rejects.toThrow(NotFoundException);
+    });
+
+    it('When user is not part of team, then error is throw', async () => {
+      jest
+        .spyOn(teamRepository, 'getTeamById')
+        .mockResolvedValue(newWorkspaceTeam());
+      jest.spyOn(teamRepository, 'getTeamUser').mockResolvedValue(null);
+
+      await expect(
+        service.changeUserRole('workspaceId', 'teamId', {
+          userId: 'useriD',
+          role: WorkspaceRole.MEMBER,
+        }),
+      ).rejects.toThrow(NotFoundException);
+    });
+
+    it('When user does not exist, then error is throw', async () => {
+      jest
+        .spyOn(teamRepository, 'getTeamById')
+        .mockResolvedValue(newWorkspaceTeam());
+      jest.spyOn(teamRepository, 'getTeamUser').mockResolvedValue(
+        new WorkspaceTeamUser({
+          id: '',
+          teamId: '',
+          memberId: '',
+          createdAt: new Date(),
+          updatedAt: new Date(),
+        }),
+      );
+      jest.spyOn(userRepository, 'findByUuid').mockResolvedValue(null);
+
+      await expect(
+        service.changeUserRole('workspaceId', 'teamId', {
+          userId: 'userId',
+          role: WorkspaceRole.MEMBER,
+        }),
+      ).rejects.toThrow(BadRequestException);
+    });
+
+    it('When user is member is updated to manager, then team manager is updated correctly', async () => {
```

Comment:
"When a member of the team is upgrade to manager, then it works" could be better